### PR TITLE
Fix link rendering (#465)

### PR DESCRIPTION
**Pull request by chris-02**
This commit should fix the issue described in #465 by disabling the fractional style for links.

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
Can you add a test?

File: tests/typ/bugs/link-rendering.typ [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#set page(width: 21em)
+#link("https://example.com/1234567890/1234567890")
```
Comment (SUPERCILEX) [inline tests/typ/bugs/link-rendering.typ:2]: Needs `#set text(fractions: true)`

Comment (SUPERCILEX) [reply tests/typ/bugs/link-rendering.typ]: Yup, thanks!

**Comment by PgBiel** [comment]
I think this should only be applied if the body and the dest are the same (i.e. ensure the body is the link before changing it). If the body is customized, then there might be legitimate situations where you want to insert a fraction in a link's label.

**Comment by PgBiel** [comment]
URL checks could work, but this can also happen for other URL/path-like things. Either way, I think this behavior (of disabling fractions) should at least have some sort of toggle (e.g. `disable-fractions: ...`) so that it is configurable, and the behavior could be enabled by default.

**Comment by laurmaedje** [comment]
I think a `disable-fractions` toggle on links is a bit much. After all you can just do `#show link: set text(fractions: false)`. In general, I think the OpenType fractions feature is somewhat questionable because it has many false positives. Enabling it for the whole document is probably a bad idea.

**Comment by PgBiel** [comment]
> I think a `disable-fractions` toggle on links is a bit much. After all you can just do `#show link: set text(fractions: false)`.

That's fair, though this PR would impose a default here. This should still work to re-enable it, right? (that is, `#show link: set text(fractions: true)`)
That's something important to test with this PR.

**Comment by PgBiel** [comment]
> Yes, if it is done with `Finalize` as done for heading configuration in `OutlineElem`, for instance. No, if done in `Show`.

Then I'm in favor of this PR moving the implementation to the `Finalize` step.

(Btw, does the same apply to `Layout` instead of `Show`? Out of curiosity.)

**Comment by laurmaedje** [comment]
> Btw, does the same apply to Layout instead of Show? Out of curiosity

Yes. But `Finalize` is generally more of a hack and should be replaced by a cleaner way to do default show-set rules as part of the realization & styling rework.

**Comment by laurmaedje** [comment]
> Then I'm in favor of this PR moving the implementation to the Finalize step.

Possibly the hyphenation setting and the linking itself should also happen in `Finalize`? Then it will _always_ happen, even if you write a show rule from scratch. I think it was like that a while ago, but I don't remember why I changed it.

**Comment by PgBiel** [comment]
> Oh, and do you want me to link to this discussion or mention you in the last commit @PgBiel?

No need! But that's your call of course.

**Comment by PgBiel** [comment]
> Maybe the [description](https://github.com/typst/typst/blob/6ad14f55d5a1b5ffef567bf1d63bb2956551bed9/src/model/realize.rs#L185) of the `Finalize` trait should be updated. It seems (if I understand both correctly) to contradict your statement above.

Yeah the description seems to be a little weird there, based on the comments here haha

**Comment by PgBiel** [comment]
Also, it seems that your tests are failing, please take a look!

**Comment by PgBiel** [comment]
> I noticed the test failing but am unable to reproduce it. I tried running tests on a fresh clone and even on a different computer but to no avail. I'll try a few more things today.

That's odd... I wonder if it could have been some one-off issue with CI. Could you try applying another commit to force a re-run? Maybe changing one letter then reverting it. Or fixing some typo or something.

**Pull request closed without merging**